auth-proposal/README.md

```diff
@@ -0,0 +1,69 @@
+# Introduction
+
+We are moving infrastructure specific logic (converting devfile concepts to k8s objects and applying them onto cluster) to devworkspace controller to provide more Kubernetes native way.
+As the result, we see devworkspace operator independent from Che which means users can run devworkspace instances via other kubernetes clients, like kubectl/oc, k8s/OpenShift Dashboards probably in less convenient way as Che is able to provide.
+
+The big piece of devworkspace logic is a routing - how user access endpoints(how he authenticates authorizes). We need some solution where user is able to use devworkspaces endpoints with and without Che Operator installed.
+So, it's the proposal we have as a result of investigation.
+You can take a look more alternatives we considered [here](https://github.com/sleshchenko/devworkspace-operator/tree/authentication-schemas/authentication-schemas). Be ready that they are not simple and detailed enough =)
+
+# Kubernetes
+
+The difficulty we have on kubernetes and workspaces authentication comes mainly from requirement Creator Access Only*.
+
+\* Creator Access Only is a feature that currently is not provided by Che at all. It supposes that we guarantee that nobody except workspace creator won't get container access inside container(neither via K8s API or workspace HTTP endpoints). It was set as a requirement for Web Terminal since we store user's token on file system. AFAIK In Che case we don't store cluster token unless user put it manually on the file system.
+
+So, we propose not to satisfy that requirement on any Kubernetes cluster. Then we can propose the following abilities for authentication:
+
+## Basic HTTPs Routing
+
+It means that we don't provide any authentication on the workspace endpoints. Everything that we need here to support it: implement/provision passthrough proxy container into workspace pod, to make workspace server available since they are supposed to listen to 127.0.0.0 only.
+
+* for simplification we can rely on default certificates configured on the Ingress controller side if we have access to CA;
+
+### Basic Authentication HTTPs Routing
```

Comment:
It's a simple solution but UX is not really great. Alternatively we are going to investigate a bit more OpenID Connect https://github.com/devfile/devworkspace-operator/issues/174 which could be used k8s cluster where it's enabled.